crates/nix_rs/src/store.rs

```diff
@@ -106,19 +104,20 @@ impl NixStoreCmd {
         }
     }
 
-    /// Given a derivation path, this function recursively queries and return all
+    /// Given the derivation paths, this function recursively queries and return all
     /// of its dependencies in the Nix store.
     pub async fn nix_store_query_requisites_with_outputs(
         &self,
-        drv_path: PathBuf,
-    ) -> Result<Vec<StorePath>, NixStoreCmdError> {
+        drv_paths: HashSet<PathBuf>,
```

Comment:
This should take a slice (`&[Pathbuf]`). There's not reason for it to require a HashSet. A convenience of a single caller should not dictate API of some core function.